x-pack/plugin/rank-text-similarity/build.gradle

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+apply plugin: 'elasticsearch.internal-es-plugin'
+apply plugin: 'elasticsearch.internal-cluster-test'
+apply plugin: 'elasticsearch.internal-yaml-rest-test'
+
+esplugin {
+  name 'rank-text-similarity'
+  description 'Text similarity reranking in search.'
+  classname 'org.elasticsearch.xpack.rank.textsimilarity.TextSimilarityRankPlugin'
+  extendedPlugins = ['x-pack-core', 'x-pack-inference']
+}
+
+dependencies {
+//  compileOnly project(path: xpackModule('core'))
```

Comment:
I'm not familiar with this DSL, so I probably made some mistakes here.
- What is the difference between `project('...')` and `project(path: xpackModule(...))`?
- When should I use `testArtifact(project())` vs just `project()` under `testImplementation`?
- What is `clusterModules` for?

Reply:
You need to use the `project(path:...)` variant. That comes from the `DependencyHandler` in gradle. The other one grabs a project object from Gradle, but that's a different thing.

`testArtifact` is for when depending on the test classes from another project. That other project has to be setup specially to export its test artifact.

`clusterModules` is for the "new" test clusters framework. This ties another project as a module that will be built, for use by java rest tests. It's only available when you have applied the java rest test plugin, ie `apply plugin: 'elasticsearch.internal-java-rest-test'` at the top of `build.gradle`.